### ISSUE TITLE
[UX] Add nowrap CSS to field suffixes (ex the machine name span) 

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#1349]
Here's how it works currently:

![backdrop-machine_name_wraps_per_word](https://cloud.githubusercontent.com/assets/2423362/11147057/72649466-8a67-11e5-8626-5cd5467ac17d.gif)

Here's how it would look if we added `white-space: nowrap;` to the `span.field-suffix`:

![backdrop-machine_name_nowrap](https://cloud.githubusercontent.com/assets/2423362/11147099/b67209b8-8a67-11e5-9558-ef0747690870.gif)

**Comment by klonos** [comment]
...just came across this again:

![backdrop-wrap_machine_name_as_a_whole](https://cloud.githubusercontent.com/assets/2423362/11924157/046d34a6-a801-11e5-831f-2986317d596f.gif)

...and almost filed a duplicate (had forgotten about filing this issue here). Saved by the auto-complete that revealed this issues title.

**Comment by jenlampton** [comment]
:+1:

**Comment by BWPanda** [comment]
PR: https://github.com/backdrop/backdrop/pull/3060

**Comment by klonos** [comment]
Hey @BWPanda ...I tried to fix this in my PR for #2731, but as @jenlampton [pointed out](https://github.com/backdrop/backdrop/pull/2783/files#r334298825), using `.field-suffix` is too generic:

> This CSS is now specific enough, it will target all suffixes for all fields on this form. If all we're after is the machine name, we should target that single item specifically.

Here's a PR that adds a `machine-name-preview` class to the `<span>`s around machine name previews, and then targets them specifically: https://github.com/backdrop/backdrop/pull/3069

Instead of repeating the same CSS for each theme, I thought that it'd be best to do it globally, and then leave it up to each contrib/custom to override as/if desired. For now, I have added the CSS to `field.css`, but let me know if it should live elsewhere (`system.css` perhaps?).

**Comment by klonos** [comment]
...or do you guys think that it'd be best to have the machine name be rendered below the field instead of to its right/left?

**Comment by herbdool** [comment]
It's unclear which is the main PR to review here.

**Comment by BWPanda** [comment]
@herbdool For the reasons mentioned in https://github.com/backdrop/backdrop-issues/issues/1349#issuecomment-583655203 I have closed my PR. Please review @klonos' instead.

**Comment by stpaultim** [comment]
The PR by @klonos seems to work for me. I tested the two examples provided in the initial description of the problem and looked for other similar pages but could not find any.

**Comment by indigoxela** [comment]
@klonos unfortunately your PR has conflicts that need to get resolved. And it really, really lags behind. So back to needs work.

**Comment by indigoxela** [comment]
The milestone doesn't fit the label, BTW. It's marked as "feature request" (why that?), but the milestone is a bugfix release.

The PR still has conflicts, unfortunately. And it hasn't been updated since Feb 2020, I don't think we should keep the milestone, anyway.

**Comment by klonos** [comment]
This doesn't add any new functionality - it simply improves the way an existing thing works. It's not clearly a bug either, so a task it is (which would justify the milestone just fine, so taking the liberty to restore it).

I have resolved the conflicts and rebased + also fixed a small coding standard issue in an adjacent line in the .css file, which I missed when I initially filed this PR.

This is ready for review/testing once again.

**Comment by indigoxela** [comment]
Many thanks for updating - it's so great to have you back. :pray:

I've one concert with `white-space: nowrap;` in general - it's not very mobile-friendly. On small screens, if the suffix is long, this may break the layout. One possibility would be to only apply when the screen isn't too small.

But I also have a concern re the hard-coded `span` tag: A field suffix might contain block level elements, or a field suffix may contain a closing tag for a field prefix. Here's an [example](https://github.com/backdrop/backdrop/blob/1.x/core/modules/image/image.field.inc#L100). My suspicion is, that your implementation would break things.

**Comment by indigoxela** [comment]
I checked the linked form in the sandbox. Yep, the PR breaks markup vadility:

![broken-markup](https://user-images.githubusercontent.com/842066/185787693-f9a216fe-7af3-4d49-822a-28a0c8319823.png)

Back to needs work.

**Comment by indigoxela** [comment]
Wait a minute... markup already *is* broken, it's just getting slightly worse now. Do we have an issue for that?

**Comment by klonos** [comment]
> ... A field suffix might contain block level elements, or a field suffix may contain a closing tag for a field prefix. ...

Good catch @indigoxela 👍🏼 ...and thanks for filing that other issue too. I'll see if I can fix things so that HTML tags in pre-/suffixes don't break things.

**Comment by klonos** [comment]
> On small screens, if the suffix is long, this may break the layout.

Right. I have a couple of ideas re that too, but I will need to play a bit with this before deciding. We could try adding `white-space: nowrap;` individually to the "Machine name:" label and the actual machine name. Perhaps we should even place the machine name underneath the field instead of next to it. Would that be too radical?

**Comment by klonos** [comment]
@indigoxela the issue with the invalid markup could be just a matter of using [`#prefix`](https://docs.backdropcms.org/form_api#prefix) vs. [`#field_prefix`](https://docs.backdropcms.org/form_api#field_prefix) and [`#suffix`](https://docs.backdropcms.org/form_api#suffix) vs. [`#field_suffix`](https://docs.backdropcms.org/form_api#fiel_suffix) properly. I often forget that these things even exist, so chances are that others make the same mistake + then people that review code also miss it.

Perhaps we need a best-practice policy, where we discourage using markup in `#prefix`/`#suffix` and use the respective `#field_*` variants for that instead(?) ...or the other way around 🤷🏼
